wallet crashes at first start

Install wallet 0.5.3 in Android 6.01; Start after install.
App asks for permissions to access media and Photos; Permissions granted.
App crashes
java.lang.RuntimeException: Failure delivering result ResultInfo{who=@android:requestPermissions:, request=1, result=-1, data=Intent { act=android.content.pm.action.REQUEST_PERMISSIONS (has extras) }} to activity {com.m2049r.xmrwallet/com.m2049r.xmrwallet.LoginActivity}: java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
	at android.app.ActivityThread.deliverResults(ActivityThread.java:3744)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:3787)
	at android.app.ActivityThread.access$1500(ActivityThread.java:153)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1424)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:5529)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:739)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:629)
Caused by: java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
	at android.support.v4.app.FragmentManagerImpl.checkStateLoss(FragmentManager.java:1842)
	at android.support.v4.app.FragmentManagerImpl.enqueueAction(FragmentManager.java:1860)
	at android.support.v4.app.BackStackRecord.commitInternal(BackStackRecord.java:650)
	at android.support.v4.app.BackStackRecord.commit(BackStackRecord.java:609)
	at com.m2049r.xmrwallet.LoginActivity.startLoginFragment(LoginActivity.java:286)
	at com.m2049r.xmrwallet.LoginActivity.onRequestPermissionsResult(LoginActivity.java:271)
	at android.app.Activity.dispatchRequestPermissionsResult(Activity.java:6634)
	at android.app.Activity.dispatchActivityResult(Activity.java:6512)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:3740)
	... 9 more

--- a/app/src/main/java/com/m2049r/xmrwallet/LoginActivity.java
+++ b/app/src/main/java/com/m2049r/xmrwallet/LoginActivity.java
@@ -593,7 +593,7 @@ public class LoginActivity extends AppCompatActivity
                 // If request is cancelled, the result arrays are empty.
                 if (grantResults.length > 0
                         && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-                    startLoginFragment();
+                    startLoginFragment = true;
                 } else {
                     String msg = getString(R.string.message_strorage_not_permitted);
                     Log.e(TAG, msg);
@@ -602,6 +602,17 @@ public class LoginActivity extends AppCompatActivity
                 }
                 break;
             default:
+        }
+    }
+
+    private boolean startLoginFragment = false;
+
+    @Override
+    protected void onResumeFragments() {
+        super.onResumeFragments();
+        if (startLoginFragment) {
+            startLoginFragment();
+            startLoginFragment = false;
         }
     }
 

--- a/app/src/main/java/com/m2049r/xmrwallet/WalletActivity.java
+++ b/app/src/main/java/com/m2049r/xmrwallet/WalletActivity.java
@@ -612,6 +612,16 @@ public class WalletActivity extends AppCompatActivity implements WalletFragment.
         getWallet().disposePendingTransaction();
     }
 
+    private boolean startScanFragment = false;
+
+    @Override
+    protected void onResumeFragments() {
+        super.onResumeFragments();
+        if (startScanFragment) {
+            startScanFragment();
+            startScanFragment = false;
+        }
+    }
 
     private void startScanFragment() {
         Bundle extras = new Bundle();
@@ -703,7 +713,7 @@ public class WalletActivity extends AppCompatActivity implements WalletFragment.
                 // If request is cancelled, the result arrays are empty.
                 if (grantResults.length > 0
                         && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-                    startScanFragment();
+                    startScanFragment = true;
                 } else {
                     String msg = getString(R.string.message_camera_not_permitted);
                     Log.e(TAG, msg);

